Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -1436,18 +1437,12 @@ namespace xr
         XrFrameWaitInfo frameWaitInfo{ XR_TYPE_FRAME_WAIT_INFO };
         XrFrameState frameState{ XR_TYPE_FRAME_STATE };
 
-        XrSecondaryViewConfigurationFrameStateMSFT secondaryViewConfigFrameState{XR_TYPE_SECONDARY_VIEW_CONFIGURATION_FRAME_STATE_MSFT};
-        std::vector<XrSecondaryViewConfigurationStateMSFT> secondaryViewConfigStates;
-        if (sessionImpl.HmdImpl.SupportedSecondaryViewConfigurationTypes.size() > 0)
+        const auto& supportedSecondaryViewConfigTypes = m_impl->sessionImpl.HmdImpl.SupportedSecondaryViewConfigurationTypes;
+        const auto secondaryViewConfigCount = supportedSecondaryViewConfigTypes.size();
+        std::vector<XrSecondaryViewConfigurationStateMSFT> secondaryViewConfigStates(secondaryViewConfigCount, { XR_TYPE_SECONDARY_VIEW_CONFIGURATION_STATE_MSFT });
+        if (secondaryViewConfigCount > 0)
         {
-            uint32_t viewCount = 0;
-            for (const auto& viewConfigurationType : sessionImpl.HmdImpl.SupportedSecondaryViewConfigurationTypes)
-            {
-                const auto& viewState = renderResources.ResourceMap.at(viewConfigurationType).ViewState;
-                viewCount += static_cast<uint32_t>(viewState.Views.size());
-            }
-
-            secondaryViewConfigStates.resize(viewCount, { XR_TYPE_SECONDARY_VIEW_CONFIGURATION_STATE_MSFT });
+            XrSecondaryViewConfigurationFrameStateMSFT secondaryViewConfigFrameState{XR_TYPE_SECONDARY_VIEW_CONFIGURATION_FRAME_STATE_MSFT};
```

Comment:
Here as well.  The variable definition need to be moved out of the if block and keep it alive when calling the xr function below.